src/app/charts/charts.component.ts

```diff
@@ -0,0 +1,64 @@
+import { Component, OnInit, Input, Output, EventEmitter } from "@angular/core";
+import { ChartType, ChartDataSets } from "chart.js";
+import { Label } from "ng2-charts";
+import { CountryInfo } from "../interface/interface";
+
+@Component({
+  selector: "app-charts",
+  templateUrl: "./charts.component.html",
+  styleUrls: ["./charts.component.scss"],
+})
+export class ChartsComponent implements OnInit {
+  @Input() chartData: CountryInfo[];
+  @Input() type: string;
+  @Input() labels: string[];
+  @Input() numberOfData: number;
+
+  activeSort: string = "";
+
+  lineChartLabels: Label[] = [];
+  lineChartType: ChartType = "line";
+  lineChartData: ChartDataSets[] = [];
+  lineChartColors = [
+    {
+      backgroundColor: "rgba(240, 52, 52, 0.8)",
+    },
+    {
+      backgroundColor: "rgba(0,255,0,0.8)",
+    },
+    {
+      backgroundColor: "rgba(0,0,0,0.8)",
+    },
+  ];
+
+  constructor() {}
+
+  ngOnInit(): void {
+    this.loadChartData(this.labels[0]);
+  }
+
+  onClick(type: string) {
+    this.lineChartLabels = [];
+    this.lineChartData = [];
+    this.loadChartData(type);
+  }
+
+  loadChartData(type: string) {
+    this.activeSort = type;
+
+    for (const label of this.labels) {
+      this.lineChartData.push({ data: [], label });
```

Comment:
Could we use Array.map here? It would be more readable, and also - it would return a new array, preventing immutability problems